### ISSUE TITLE
Debugger calls AD7Engine Finalized too soon

### ISSUE THREAD

**Issue by mousetraps** [Microsoft/nodejstools#823]
_Copied from @purell https://github.com/Microsoft/nodejstools/issues/246#issuecomment-194502140_

@mousetraps : I did what you said about the "Diagnostic Info" windows and this is what I got when the debugger doesn't start : 

```
[20:48:26.2699168] --------------------------------------------------------------------------------
[20:48:26.2769168] AD7Engine Created (25640749)
[20:48:26.2769168] AD7Engine Called SetRegistryRoot
[20:48:26.2769168] AD7Engine Called SetLocale
[20:48:26.2769168] AD7Engine LaunchSuspended Called with flags 'LAUNCH_ENABLE_ENC' (25640749)
[20:48:27.0039168] AD7Engine Finalized (25640749)
```

Normaly before the Finalized line we are suppose to the the debugger trying to connect (Debugger connecting to URI: tcp://localhost:....) but the log stop there and the action too.

By the way, is it helping you more ?

My Configuration :
  - Windows 7 64 Pro
  - Visual Studio PRO Ultimate 2013
  - Node v0.12.0
  - Last NTVS installled, from https://github.com/Microsoft/nodejstools/releases/download/v1.1.Dev-3.1.2016/NTVS.Dev.2016-03-01.VS.2013.msi

This problem is more than annoying now, I have to relaunch Visual Studio 2013 a lot of time per hours while developing. I think that in a day I can relaunch it 100 times ! Even if I don't like Webstorm, I think that I will come back to it very shortly :(

Maybe we should re-open the case ?

**Comment by purell** [comment]
I don't know if it could help but I notice that If I stop the debugger using Visual Studio menu [Debug + Stop Debugging] the problem occurred more frequently than stopping the program by closing the console. If I'm only closing the console each time the problem occur rarely.

**Comment by mjbvz** [comment]
@purell To unblock yourself, try using VS 2015 Update 2 and NTVS 1.2 Alpha. I've tried to repo the issue using the latest versions and Node v0.12.0, but was not able to.

Either way, I think we should add more diagnostic logging to the AD7Engine so we can better investigate this sort of problem.

**Comment by mjbvz** [comment]
Closing. @purell please give our most recent releases a try and let me know if you are still seeing this issue.

Thanks.

**Comment by Ampakinetic** [comment]
Hi, I've got this same issue and it's always been like this on my PC (since I started using Node Tools quite a long time ago), it seems to be worse with the 1.2 release.   My config is below and this is the diagnostics log when the debugger refuses to start.

[08:00:38.5124147] AD7Engine Created (1316945)
[08:00:38.5124147] AD7Engine Called SetRegistryRoot
[08:00:38.5124147] AD7Engine Called SetLocale
[08:00:38.5129148] AD7Engine LaunchSuspended Called with flags 'LAUNCH_ENABLE_ENC' (1316945)
[08:00:38.7894499] AD7Engine Finalized (1316945)
----

Microsoft Visual Studio Community 2015
Version 14.0.25421.03 Update 3
Microsoft .NET Framework
Version 4.6.01055

Installed Version: Community

Visual Basic 2015   00322-20000-00000-AA439
Microsoft Visual Basic 2015

Visual C# 2015   00322-20000-00000-AA439
Microsoft Visual C# 2015

Visual C++ 2015   00322-20000-00000-AA439
Microsoft Visual C++ 2015

Application Insights Tools for Visual Studio Package   7.1.00719.1
Application Insights Tools for Visual Studio

ASP.NET and Web Tools 2015.1 (Beta8)   14.1.11107.0
ASP.NET and Web Tools 2015.1 (Beta8)

ASP.NET Web Frameworks and Tools 2012.2   4.1.41102.0
For additional information, visit http://go.microsoft.com/fwlink/?LinkID=309563

ASP.NET Web Frameworks and Tools 2013   5.2.40314.0
For additional information, visit http://www.asp.net/

Common Azure Tools   1.8
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Fix File Encoding   1.2
Prevents Visual Studio from adding BOM to UTF-8 files.

GitHub.VisualStudio   1.0
A Visual Studio Extension that brings the GitHub Flow into Visual Studio.

JavaScript Language Service   2.0
JavaScript Language Service

JavaScript Project System   2.0
JavaScript Project System

Microsoft Azure Mobile Services Tools   1.4
Microsoft Azure Mobile Services Tools

Node.js Tools   1.2.40726.00
Adds support for developing and debugging Node.js apps in Visual Studio

Node.js Tools - Profiling   1.2.40726.00
Profiling support for Node.js projects.

NuGet Package Manager   3.4.4
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

PreEmptive Analytics Visualizer   1.2
Microsoft Visual Studio extension to visualize aggregated summaries from the PreEmptive Analytics product.

SQL Server Data Tools   14.0.60519.0
Microsoft SQL Server Data Tools

TypeScript   1.8.34.0
TypeScript tools for Visual Studio

Visual Studio Tools for Apache Cordova   Update 10
Visual Studio Tools for Apache Cordova

Visual Studio Tools for Universal Windows Apps   14.0.25420.01
The Visual Studio Tools for Universal Windows apps allow you to build a single universal app experience that can reach every device running Windows 10: phone, tablet, PC, and more. It includes the Microsoft Windows 10 Software Development Kit.

VisualSVN   5.1.3
Integration with Subversion version control. For more information about VisualSVN, see the VisualSVN website at http://www.visualsvn.com
Copyright © 2005-2015 VisualSVN Ltd. All rights reserved.

Xamarin   4.1.0.530 (2e39740)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin.Android   6.1.0.71 (4e27558)
Visual Studio extension to enable development for Xamarin.Android.

Xamarin.iOS   9.8.0.323 (39ebb77)
Visual Studio extension to enable development for Xamarin.iOS.
